common/common_structs/src/wrapper_types.rs

```diff
@@ -92,3 +94,50 @@ impl<M: ManagedTypeApi> RawResultWrapper<M> {
         result
     }
 }
+
+pub struct PaymentsWrapper<M: ManagedTypeApi> {
+    payments: PaymentsVec<M>,
+}
+
+impl<M: ManagedTypeApi> PaymentsWrapper<M>
+where
+    M: ManagedTypeApi + SendApi,
+{
+    pub fn new() -> Self {
+        Self {
+            payments: ManagedVec::new(),
+        }
+    }
+
+    pub fn new_from_payments(payments: PaymentsVec<M>) -> Self {
+        Self { payments }
+    }
+
+    pub fn push(&mut self, payment: EsdtTokenPayment<M>) {
+        if payment.amount > 0 {
+            self.payments.push(payment);
+        }
+    }
+
+    pub fn send_to_address<T: SendApi>(&self, address: &ManagedAddress<M>) {
```

Comment:
Remove `T` and just use `M`, it already has the `SendApi` bound.

Reply:
Updated.